### PR TITLE
Dev/guma/polyline implements icurve

### LINKED ISSUE #263: Polyline implements ICurve
IMPLEMENT:
- [x] PointAt(t)
- [x] PointAtLength(d)
- [x] TangentAt(t)
- [x] TangentAtLength(d)
- [x] ClosestPoint(pt)
- [x] ClosestParameter(pt)
- [x] LengthAt(t)
- [x] ParameterAt(d len)
- [x] GetBoundingBox()
- [x] Length
- [x] StartPoint
- [x] EndPoint

REMOVE:
- [x] public int Degree { get; }
- [x] public List ControlPointLocations { get; }
- [x] public List ControlPoints { get; }
- [x] public KnotVector Knots { get; }
- [x] public Interval Domain { get; }

### PR DESCRIPTION
**Pull request by d3ssy**
### What type of PR is this? (check all applicable)

- [x] ♻️ Refactor

### Description

This PR implements ICurve methods on Polyline.

### Related Tickets & Documents
Closes #263 

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [x] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/GShark/Geometry/Polyline.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.InteropServices;
```
Comment (cesarecaoduro) [inline src/GShark/Geometry/Polyline.cs:6]: Why this?

File: src/GShark/Operation/Offset.cs [hunk 2]
```diff
@@ -100,7 +100,7 @@ public static Polyline Polyline(Polyline poly, double distance, Plane pln)
                     continue;
                 }
 
-            Intersection:
+                Intersection:
```
Comment (cesarecaoduro) [inline src/GShark/Operation/Offset.cs:103]: Maybe just clean spaces

**Review by cesarecaoduro — COMMENTED** [review]
minor comments

File: src/GShark.Test.XUnit/Operation/OffsetTests.cs [hunk 3]
```diff
@@ -61,20 +61,23 @@ public void Returns_The_Offset_Of_A_Open_Polyline()
...
-            Polyline offsetResult = Offset.Polyline(pl, offset, Plane.PlaneXY);
+            Polyline offsetPolygon = Offset.Polyline(polygon, offset, Plane.PlaneXY);
 
             // Assert
-            offsetResult[0].DistanceTo(offsetResult[offsetResult.Count - 1]).Should().Be(0.0);
-            Point3 pt = offsetResult.PointAt(0.5);
-            Point3 closestPt = pl.ClosestPoint(pt);
-            pt.DistanceTo(closestPt).Should().BeApproximately(offset, GSharkMath.MinTolerance);
+            polygon[vertex].DistanceTo(offsetPolygon[vertex]).Should()
+                .BeApproximately(expectedDistance, GSharkMath.MaxTolerance);
```
Comment (sonomirco) [inline src/GShark.Test.XUnit/Operation/OffsetTests.cs:80]: Doesn't pass with MinTolerance?

**Comment by sonomirco** [comment]
We know we have to unify the documentation between the different types of curves and the method into ``ICurve``

File: src/GShark.Test.XUnit/Operation/OffsetTests.cs (hunk 3 above)
Comment (d3ssy) [reply src/GShark.Test.XUnit/Operation/OffsetTests.cs:80]: Hahahaha... I was still thinking in the old version

File: src/GShark/Operation/Offset.cs (hunk 2 above)
Comment (d3ssy) [reply src/GShark/Operation/Offset.cs:117]: Weird... in the file that indentation isn't there

**Pull request merged**